### PR TITLE
Consume updated common-core to fix issues around ip spaces

### PR DESCRIPTION
**Pull request by rocknes**
Recently two bugs related to ip-spaces were fixed in CPI (common core). viz.
1. CPI not supporting data center group org vdc networks for ip space related operations
2. CPI erroring out while processing org vdc networks that have more than one uplink

Fix for both bugs were committed to common-core. This PR updates the version of common-core consumed by CAPVCD, to get the fixes.

Testing Done:
Setup a CSE testbed where the following VCD objects were present
1. A data center group org vdc network
2. An Ip Space that is powering the org vdc network
3. An external network connected to the same org vdc network

Built CPI and CAPVCD images with the fix and hosted them on a private registry.
Updated CSE server config to consume the updated dev images.
Created a cluster from CSE UI.

Outcome:
CSE cluster was created successfully along with the load balancer.
Verified IP was reserved for the cluster load balancer properly in the IP space.

- 

## Checklist
- [X] tested locally
- [ ] updated any relevant dependencies
- [ ] updated any relevant documentation or examples

## API Changes
Are there API changes?
- [ ] Yes
- [X ] No

If yes, please fill in the below

1. Updated conversions?
    - [ ] Yes
    - [X] No
    - [ ] N/A
3. Updated CRDs?
    - [ ] Yes
    - [X] No
    - [ ] N/A
4. Updated infrastructure-components.yaml?
    - [ ] Yes
    - [] No
    - [X] N/A
5. Updated `./examples/capi-quickstart.yaml`?
    - [ ] Yes
    - [] No
    - [X] N/A
6. Updated necessary files under `./infrastructure-vcd/v1.0.0/`?
   - [ ] Yes
   - [ ] No
   - [X] N/A

- - -
This change is [Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/651)

### CODE REVIEW

**Review by lzichong — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/vmware/cluster-api-provider-cloud-director
 
-go 1.20
```
Comment (arunmk) [inline go.mod:3]: This is a go version update. Sometimes this needs a change to build scripts. Do we need this update or was this accidental?

**Review by arunmk — APPROVED** [review]
LGTM with a go version question. I think Adnan may be able to answer about the script changes needed.

File: go.mod (hunk 1 above)
Comment (rocknes) [reply go.mod:3]: This came is a part of common core. Common core has been at go 1.21 for like 4 months now

**Review by arunmk — APPROVED** [review]
Looks good. Traditionally we have used a different commit to move to the new version, but it should be fine in this case.

**Pull request merged**